NetSDK/Services/Impressions/Classes/InMemoryTreatmentLog.cs

```diff
@@ -0,0 +1,77 @@
+﻿using log4net;
+using Splitio.Domain;
+using Splitio.Services.Impressions.Interfaces;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Splitio.Services.Impressions.Classes
+{
+    public class InMemoryTreatmentLog : ITreatmentLog
+    {
+        private ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>> cache;
```

Comment:
Let's move away from this and use a plain blocking queue. It leads to a much simpler code.

Latest Java uses a queue.